mlem/contrib/sagemaker/build.py

```diff
@@ -0,0 +1,135 @@
+import base64
+import os
+from typing import ClassVar, Optional
+
+import boto3
+import sagemaker
+from pydantic import BaseModel
+
+from ...core.objects import MlemModel
+from ...ui import EMOJI_BUILD, EMOJI_KEY, echo, set_offset
+from ..docker.base import DockerEnv, DockerImage, RemoteRegistry
+from ..docker.helpers import build_model_image
+from .runtime import SageMakerServer
+
+IMAGE_NAME = "mlem-sagemaker-runner"
+
+
+class AWSVars(BaseModel):
+    """AWS Configuration"""
+
+    profile: str
+    """AWS Profile"""
+    bucket: str
+    """S3 Bucket"""
+    region: str
+    """AWS Region"""
+    account: str
+    """AWS Account name"""
+    role_name: str
+    """AWS Role name"""
+
+    @property
+    def role(self):
+        return f"arn:aws:iam::{self.account}:role/{self.role_name}"
+
+    def get_sagemaker_session(self):
+        return sagemaker.Session(
+            self.get_session(), default_bucket=self.bucket
+        )
+
+    def get_session(self):
+        return boto3.Session(
+            profile_name=self.profile, region_name=self.region
+        )
+
+
+def ecr_repo_check(region, repository, session: boto3.Session):
+    client = session.client("ecr", region_name=region)
+
+    repos = client.describe_repositories()["repositories"]
+
+    if repository not in {r["repositoryName"] for r in repos}:
+        echo(EMOJI_BUILD + f"Creating ECR repository {repository}")
+        client.create_repository(repositoryName=repository)
+
+
+class ECRegistry(RemoteRegistry):
+    """ECR registry"""
+
+    class Config:
+        exclude = {"aws_vars"}
+
+    type: ClassVar = "ecr"
+    account: str
+    """AWS Account"""
+    region: str
+    """AWS Region"""
+
+    aws_vars: Optional[AWSVars] = None
+    """AWS Configuration cache"""
+
+    def login(self, client):
```

Comment:
can put this in try/catch, due to possible index error OR retrieval error for the authorisation token